lucet-runtime/lucet-runtime-internals/src/instance.rs

```diff
@@ -1471,6 +1471,29 @@ impl TerminationDetails {
             _ => None,
         }
     }
+    /// Try to interpret the termination details as a provided exit code.
+    ///
+    /// The most consistent form of `TerminationDetails::Provided` comes from Lucet's
+    /// implementation of `proc_exit`, which exits with a `Provided` holding the given exit code.
+    /// For cases where a Lucet user simply wants "`proc_exit` or continue panicking" behavior,
+    /// `as_exitcode` can simplify handling `TerminationDetails`.
+    pub fn as_exitcode(&self) -> Option<u32> {
+        match self {
+            TerminationDetails::Provided(a) => {
+                // I apologize for this load-bearing `as u32`.
+                // Wasi uses an u32 for the proc_exist status (`lucet_wasi::Exitcode`) in the
+                // witx. However, wasmtime::Trap exit status is an i32, so the
+                // wiggle::Trap::I32Exit variant mirrors Wasmtime. The `as u32` lets this method
+                // return a type equivalent to `lucet_wasi::Exitcode`, but users interested in the
+                // full range of `wiggle::Trap` will have to handle an i32 variant.
+                match a.downcast_ref::<wiggle::Trap>() {
```

Comment:
and this causes a `wiggle` dependency in `lucet_runtime_internals`. i think that's _okay_, but it's a weird spot where the runtime knows more about the bindings just for a types reason.

Reply:
the other way to solve this problem would be to define a lucet_runtime_internals::Trap that has the same structure (it is, after all, pretty trivial) and have the lucet-wiggle boundary convert between them. We could then enforce that TerminationDetails is always a lucet_runtime::Trap as well, since this Any business is, in my opinion, not really worth the trouble. the wiggle Trap is designed to mimic wasmtime's and this would make lucet more like wasmtime, which I think is a win.